### DISCUSSION TITLE
Support for Object-Oriented Programming

### DISCUSSION THREAD

**Discussion by rrisner** [SaptakBhoumik/Peregrine#15] [Ideas]
Are classes an intended feature for Swallow? I think it would be difficult for some users to adopt the language without support for object-oriented design patterns. Is it possible for the language to support C++ code and classes in the same way as (or instead of) C code? 

I think that "A compiled and typed Python-like language which runs nearly as fast as C++" sounds like it could be compelling for many use cases, including data science. It could bridge the gap between the ease of writing for interpreted languages and the performance of tedious C++ code. It could potentially also solve one problem I have with Python in that if I make a small application, I cannot share it with anyone who does not have Python installed. The down-side to that is that it would also be easier to write and share malicious code. Compile-time may also increase as more conversions are done between Swallow and C/C++.

**Comment by SaptakBhoumik** [comment]
@rrisner Thank you for your question. Well yes, it will have classes. Considering the fact that objects are just glorified structs it won't be that difficult

**Reply by delioda79** [reply]
Are you talking about actual old style classes or you thinking to take the approach Go and Rust (and V) took? I just think the second one is a better approach and as we know helps with forcing to use composition.

**Reply by SaptakBhoumik** [reply]
We have not decided to be honest

**Comment by yam-1111** [comment]
This could be python 4.x

**Comment by Vbansal21** [comment]
Maybe consider taking a look at how 'Julia' language implemented multiple dispatch, it is like classes but with speed of c.

**Comment by SaptakBhoumik** [comment]
Just a small update
Peregrine now supports OOP similar to python
